lib/src/Base/Stat/IsotropicCovarianceModel.cxx

```diff
@@ -0,0 +1,235 @@
+#include "openturns/CovarianceModel.hxx"
+#include "openturns/CovarianceModelImplementation.hxx"
+
+BEGIN_NAMESPACE_OPENTURNS
+
+CLASSNAMEINIT(IsotropicCovarianceModel)
+
+
+/* Parameters constructor */
+IsotropicCovarianceModel::IsotropicCovarianceModel(const CovarianceModel & oneDimensional,
+  const UnsignedInteger inputDimension)
+  : CovarianceModelImplementation()
+{
+  setKernel(oneDimensional);
+  setInputDimension(inputDimension);
+  setActiveParameter(kernel_.getActiveParameter());
+  setOutputCorrelation(kernel_.getOutputCorrelation());
+  setScale(kernel_.getScale());
+  setAmplitude(kernel_.getAmplitude());
+  setNuggetFactor(kernel_.getNuggetFactor());
+}
+
+/* Input dimension accessor */
+void IsotropicCovarianceModel::setInputDimension(const UnsignedInteger inputDimension)
+{
+  if (inputDimension < 1)
+  {
+    throw InvalidArgumentException(HERE) << "In IsotropicCovarianceModel::setInputDimension, input dimension should be > 0.";
+  }
+  inputDimension_ = inputDimension;
+}
+
+/* 1D covariance kernel accessor */
+void IsotropicCovarianceModel::setKernel(const CovarianceModel & oneDimensional)
+{
+  // Check if the given stationary covariance kernel has both input and output dimension equal to 1.
+  const UnsignedInteger kernelInputDimension = oneDimensional.getInputDimension();
+  if (kernelInputDimension != 1)
+  {
+    throw InvalidArgumentException(HERE) << "In IsotropicCovarianceModel::setKernel, the supplied covariance kernel must have input dimension 1, not " << kernelInputDimension <<".";
+  }
+  const UnsignedInteger kernelOutputDimension = oneDimensional.getOutputDimension();
+  if (kernelOutputDimension != 1)
+  {
+    throw InvalidArgumentException(HERE) << "In IsotropicCovarianceModel::setKernel, the supplied covariance kernel must have output dimension 1, not " << kernelOutputDimension <<".";
+  }
+  outputDimension_ = kernelOutputDimension;
+  kernel_ = oneDimensional;
+}
+
+CovarianceModel IsotropicCovarianceModel::getKernel() const
+{
+  return kernel_;
+}
+
+/* Virtual constructor */
+IsotropicCovarianceModel * IsotropicCovarianceModel::clone() const
+{
+  return new IsotropicCovarianceModel(*this);
+}
+
+/* Computation of the covariance function */
+SquareMatrix IsotropicCovarianceModel::operator() (const Point & s,
+  const Point & t) const
+{
+  return (*this)(t - s);
+}
+
+SquareMatrix IsotropicCovarianceModel::operator() (const Point & tau) const
+{
+  // for the time being, the output dimension is necessarily 1.
+  SquareMatrix result(1);
+  result(0, 0) = computeAsScalar(tau);
+  return result;
+}
+
+/* Computation of the covariance function */
+Scalar IsotropicCovarianceModel::computeAsScalar(const Point & tau) const
+{
+  if (tau.getDimension() != inputDimension_) throw InvalidArgumentException(HERE) << "Error: expected a shift of dimension=" << inputDimension_ << ", got dimension=" << tau.getDimension();
+  const Point tauNorm(1, tau.norm());
+  return kernel_.computeAsScalar(Point(1), tauNorm);
```

Comment:
```suggestion
return kernel_.computeAsScalar(tauNorm);
```

Reply:
reason is that kernel is stationnary

Reply:
When I apply your suggestion, I get the following compiler error:
```
openturns/lib/src/Base/Stat/IsotropicCovarianceModel.cxx: In member function ‘OT::Scalar OT::IsotropicCovarianceModel::computeAsScalar(const OT::Point&) const’:
openturns/lib/src/Base/Stat/IsotropicCovarianceModel.cxx:102:41: error: no matching function for call to ‘OT::CovarianceModel::computeAsScalar(const OT::Point&) const’
   return kernel_.computeAsScalar(tauNorm);
                                         ^
In file included from openturns/lib/src/Base/Stat/openturns/IsotropicCovarianceModel.hxx:23,
                 from openturns/lib/src/Base/Stat/IsotropicCovarianceModel.cxx:20:
openturns/lib/src/Base/Stat/openturns/CovarianceModel.hxx:72:18: note: candidate: ‘virtual OT::Scalar OT::CovarianceModel::computeAsScalar(const OT::Point&, const OT::Point&) const’
   virtual Scalar computeAsScalar (const Point & s,
                  ^~~~~~~~~~~~~~~
openturns/lib/src/Base/Stat/openturns/CovarianceModel.hxx:72:18: note:   candidate expects 2 arguments, 1 provided
make[2]: *** [lib/src/CMakeFiles/OT.dir/build.make:4873: lib/src/CMakeFiles/OT.dir/Base/Stat/IsotropicCovarianceModel.cxx.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:845: lib/src/CMakeFiles/OT.dir/all] Error 2
make: *** [Makefile:163: all] Error 2
```

Reply:
Indeed. `computeAsScalar(tau)` is defined at `StationaryCovarianceModel` level (class that will be dropped).
Thus for the moment, do not use the `override` in `hxx`

Reply:
So should I leave this as it is until you define `computeAsScalar(tau)` in `CovarianceModelImplementation`?

Reply:
I've added comments stating that this is a temporary workaround and I've update the TODO file so we remember this.